### PR TITLE
Made validation able to handle embedded structs

### PR DESCRIPTION
**Pull request by thomshutt**

### CODE REVIEW

File: remoteconfig_test.go [hunk 1]
```diff
@@ -591,52 +591,30 @@ func (s *RemoteConfigSuite) TestLoadConfigFromURLError() {
...
+
+func (s *RemoteConfigSuite) TestReadJSONParseEmbeddedStruct() {
+	cfgBuffer := bytes.NewBufferString(validConfigJSON)
 
 	c := &SampleConfig{}
 	err := ReadJSONValidate(cfgBuffer, c)
 	assert.Nil(s.T(), err)
+	assert.NotNil(s.T(), c.EmbeddedStr)
+	assert.EqualValues(s.T(), "abc", *c.EmbeddedStr)
+}
```
Comment (jyehbrightcove) [inline remoteconfig_test.go]: could check the embeddedint, but really i guess if that's broken, it's really more a set fire to everything situation.

Comment (jyehbrightcove) [inline remoteconfig_test.go]: Dang straight I'm a zencoder member.

Comment (thomshutt) [inline remoteconfig_test.go]: Sure! Shouldn't be a situation where only one of those two gets set but on the other hand, Computers.

**Comment by jyehbrightcove** [comment]
LGTM pending circle

**Comment by stuarthicks** [comment]
LGTM :shipit:

**Pull request merged**